pkg/sink/codec/debezium/encoder.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debezium
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"time"
+
+	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/pkg/config"
+	"github.com/pingcap/tiflow/pkg/errors"
+	"github.com/pingcap/tiflow/pkg/sink/codec"
+	"github.com/pingcap/tiflow/pkg/sink/codec/common"
+)
+
+type BatchEncoder struct {
+	keyBuf      *bytes.Buffer
+	valueBuf    *bytes.Buffer
+	callbackBuf []func()
+	batchSize   int
+
+	config *common.Config
+	codec  *Codec
+}
+
+// EncodeCheckpointEvent implements the RowEventEncoder interface
+func (d *BatchEncoder) EncodeCheckpointEvent(ts uint64) (*common.Message, error) {
+	// Currently ignored. Debezium MySQL Connector does not emit such event.
+	return nil, nil
+}
+
+// AppendRowChangedEvent implements the RowEventEncoder interface
+func (d *BatchEncoder) AppendRowChangedEvent(
+	_ context.Context,
+	_ string,
+	e *model.RowChangedEvent,
+	callback func(),
+) error {
+	value, err := d.codec.EncodeRowChangedEvent(e)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	d.valueBuf.Write(value)
+	d.batchSize++
```

Comment:
official debezium also support batch multiple events into one message?

Reply:
I see, it write the batch version.

Reply:
If we have to support batch, batch size and batched message size in bytes must be checked, to prevent batch too many small messages into a large one, and then cause the `message too large` error from the kafka.